src/FormAlter/MediaFormAlter.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\media_mpx\FormAlter;
+
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Entity\ContentEntityForm;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Messenger\MessengerTrait;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\media\Entity\Media;
+use Drupal\media_mpx\MpxLogger;
+use Drupal\media_mpx\Repository\MpxMediaType;
+use Drupal\media_mpx\Service\UpdateVideoItem\UpdateVideoItem;
+use Drupal\media_mpx\Service\UpdateVideoItem\UpdateVideoItemRequest;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Alters the media form for mpx items to add "Reimport" button.
+ *
+ * @package Drupal\media_mpx\FormAlter
+ */
+class MediaFormAlter implements ContainerInjectionInterface {
+
+  use StringTranslationTrait;
+  use MessengerTrait;
+
+  /**
+   * The mpx media types repository.
+   *
+   * @var \Drupal\media_mpx\Repository\MpxMediaType
+   */
+  private $mpxMediaTypeRepository;
+
+  /**
+   * Update Video Item service.
+   *
+   * @var \Drupal\media_mpx\Service\UpdateVideoItem\UpdateVideoItem
+   */
+  private $updateService;
+
+  /**
+   * The custom media mpx logger service.
+   *
+   * @var \Drupal\media_mpx\MpxLogger
+   */
+  private $logger;
+
+  /**
+   * MediaFormAlter constructor.
+   */
+  private function __construct(MpxMediaType $mpxMediaTypeRepository, UpdateVideoItem $updateService, MpxLogger $logger) {
+    $this->mpxMediaTypeRepository = $mpxMediaTypeRepository;
+    $this->updateService = $updateService;
+    $this->logger = $logger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('media_mpx.repository.mpx_media_types'),
+      $container->get('media_mpx.service.update_video_item'),
+      $container->get('media_mpx.exception_logger')
+    );
+  }
+
+  /**
+   * Implements hook_form_FORM_ID_alter().
+   */
+  public function alter(&$form, FormStateInterface $form_state, $form_id): void {
+    if ($this->alterAppliesToForm($form_state) === FALSE) {
+      return;
+    }
+
+    $form['actions']['reimport'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Reimport mpx data'),
+      '#submit' => [[$this, 'reimportCallback']],
+    ];
+  }
+
+  /**
+   * Callback for the 'reimport' button.
+   */
+  public function reimportCallback(array $form, FormStateInterface $formState) {
+    /* @var \Drupal\Core\Entity\ContentEntityForm $form_object */
+    $form_object = $formState->getFormObject();
+    $video = $form_object->getEntity();
+
+    if (!$id_field = $this->resolveIdFieldName($video)) {
+      $this->messenger()->addError($this->t('The mpx field @field is not mapped to any local field. The Media source must be configured
```

Comment:
I'm not sure this is possible. When the media type is created, Drupal automatically creates a "source" field. See `\Drupal\media\MediaSourceBase::createSourceField()`.

If it is somehow possible, lets hide the button in the form if a source field isn't configured.

Reply:
@deviantintegral covering for cases where the data is coming from a migration into an existing type. Field mappings might not be applied yet, and this can effectively return null. 

I'll hide the button in that case 👍

Reply:
@juampynr seems like this could be a good improvement for the migration - it should make defining a source field required I think.